doc/langref_common.md

```diff
@@ -106,28 +106,34 @@ dynasm!(ops
 
 In order to describe flow control effectively, dynasm-rs supports labels. However, since the assembly templates can be combined in a variety of ways at the mercy of the program using dynasm-rs, the semantics of these labels are somewhat different from how labels work in a static assembler.
 
-Dynasm-rs distinguishes between three different types of labels: global, local and dynamic labels. Their syntax is as follows:
+Dynasm-rs distinguishes between four different types of labels: global, local, dynamic and extern. Their syntax is as follows:
 
 Table 2: dynasm-rs label types
 
-Type    | Definition   | Reference
---------|--------------|-----------
-Local   | `label:`     | `>label` or `<label`
-GLobal  | `->label:`   | `->label`
-Dynamic | `=>expr`     | `=>expr`
-Extern  | `-`          | `extern expr`
+Type    |  Kind   | Definition   | Reference
+--------|---------|--------------|-----------
+Local   | static  | `label:`     | `>label` or `<label`
+GLobal  | static  | `->label:`   | `->label`
+Dynamic | dynamic | `=>expr`     | `=>expr`
+Extern  | dynamic | `-`          | `extern expr`
+
+Static label kinds  ("Local" and "Global") have their addresses resolved at the
```

Comment:
Is this true?

Reply:
This doesn't seem correct. All label types are resolved at `commit` time. Extern is also not a dynamic label, it's something completely different (it is literally just targetting a given address).

The difference between static and dynamic labels is that the identifier (name) is fixed at rustc compile time for static labels, while you can create new identifiers (basically just an incrementing integer) at runtime for dynamic labels.

Reply:
I must admit, there was a fair amount of guesswork going on here. I don't actually know how dynasm works under the hood!

What I'd like the docs to be able to tell me is which kinds of label can be looked up with `LabelRegistry::resolve_static()` and what kinds can be looked up with `LabelRegistry::resolve_dynamic()`.

I assume that `local` and `global` are for `resolve_static()`, and that `dynamic` is (guess what?) resolve_dynamic(). But what about `extern`?

Reply:
Sorry for the delay, life has a way.

Anyway, `extern` doesn't need a lookup. if you assemble `jmp extern 0x1234` it'll literally insert a jump to 0x1234. This is mostly relevant to x86, as the default jump in x86 can access the entire address space, always. However, that jump is relative, so we need to know about it for when the buffer gets moved around.

Reply:
I think https://github.com/CensoredUsername/dynasm-rs/pull/92/commits/a544deae8d946f5ae7366c238ca9f8e3df26ba2c addresses the above comments.